ui-v2/app/adapters/intention.js

```diff
@@ -24,14 +28,30 @@ export default Adapter.extend({
     return request`
       POST /v1/connect/intentions?${{ [API_DATACENTER_KEY]: data[DATACENTER_KEY] }}
 
-      ${serialized}
+      ${{
+        SourceNS: serialized.SourceNS,
+        DestinationNS: serialized.DestinationNS,
+        SourceName: serialized.SourceName,
+        DestinationName: serialized.DestinationName,
+        SourceType: serialized.SourceType,
+        Action: serialized.Action,
+        Description: serialized.Description,
+      }}
     `;
   },
   requestForUpdateRecord: function(request, serialized, data) {
     return request`
       PUT /v1/connect/intentions/${data[SLUG_KEY]}?${{ [API_DATACENTER_KEY]: data[DATACENTER_KEY] }}
 
-      ${serialized}
+      ${{
+        SourceNS: serialized.SourceNS,
+        DestinationNS: serialized.DestinationNS,
+        SourceName: serialized.SourceName,
+        DestinationName: serialized.DestinationName,
+        SourceType: serialized.SourceType,
+        Action: serialized.Action,
+        Description: serialized.Description,
+      }}
     `;
```

Comment:
This means we can easily prevent certain props/values from being sent in a request, and means we can get rid of our custom `writable` function. We only did it for intentions for now as we were there, but we'll probably remove elsewhere in another PR and eventually delete our custom `writable`. You'll see us removing this for the intentions model/serializer further down.

Reply:
Do you use serializer at all with the new adapter? Just curious as in out-of-the-box ember data, that's where this would be done.

Reply:
We use our old style ApplicationSerializer, but we don't normalize anything there, the serializer deals with all the HTTP header bits still but not a lot else. Long term we hope to move everything to this style of normalization as its a easier to see whats happening, and looks exactly like the Consul API documentation, um more notes here:

https://github.com/hashicorp/consul/pull/6285

We still plan on a similar refactoring of our Serializers (similar to what we did for our Adapters) for response serializing, just its not high on the priority list right now, it might happen pre-1.8 or something.